converter/internal/prometheusconvert/digitalocean.go

```diff
@@ -0,0 +1,63 @@
+package prometheusconvert
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/agent/component/discovery/digitalocean"
+	"github.com/grafana/agent/converter/diag"
+	"github.com/grafana/agent/converter/internal/common"
+	"github.com/grafana/agent/pkg/river/rivertypes"
+	"github.com/grafana/agent/pkg/river/token/builder"
+	promconfig "github.com/prometheus/common/config"
+	promdigitalocean "github.com/prometheus/prometheus/discovery/digitalocean"
+)
+
+func appendDiscoveryDigitalOcean(f *builder.File, jobName string, sdConfig *promdigitalocean.SDConfig) (discovery.Exports, diag.Diagnostics) {
+	discoveryDigitalOceanArgs, diags := toDiscoveryDigitalOcean(sdConfig)
+	common.AppendBlockWithOverride(f, []string{"discovery", "digitalocean"}, jobName, discoveryDigitalOceanArgs)
+	return newDiscoverExports("discovery.digitalocean." + jobName + ".targets"), diags
+}
+
+func toDiscoveryDigitalOcean(sdConfig *promdigitalocean.SDConfig) (*digitalocean.Arguments, diag.Diagnostics) {
+	if sdConfig == nil {
+		return nil, nil
+	}
+
+	return &digitalocean.Arguments{
+		RefreshInterval: time.Duration(sdConfig.RefreshInterval),
+		Port:            sdConfig.Port,
+		BearerToken:     rivertypes.Secret(sdConfig.HTTPClientConfig.BearerToken),
+		BearerTokenFile: sdConfig.HTTPClientConfig.BearerTokenFile,
+		ProxyURL:        config.URL(sdConfig.HTTPClientConfig.ProxyURL),
+		FollowRedirects: sdConfig.HTTPClientConfig.FollowRedirects,
+		EnableHTTP2:     sdConfig.HTTPClientConfig.EnableHTTP2,
+	}, validateDiscoveryDigitalOcean(sdConfig)
+}
+
+func validateDiscoveryDigitalOcean(sdConfig *promdigitalocean.SDConfig) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	if sdConfig.HTTPClientConfig.BasicAuth != nil {
+		diags.Add(diag.SeverityLevelWarn, "unsupported basic_auth for digitalocean_sd_configs")
```

Comment:
This applies to all the checks like this and there might already be discussion on this, but is there a reason this is `warn` instead of `error`?

Reply:
Unsupported configs are classified as warnings because we can do the conversion but those pieces are ignored. We will default to failing on warnings but allow overriding them by choice.